lib/improver/tests/psychrometric_calculations/precip_phase_probability/test_ProbPhase.py

```diff
@@ -0,0 +1,176 @@
+        self.assertDictEqual(result.attributes, self.mandatory_attributes)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_prob_rain(self):
+        """Test that process returns a cube with the right name, units and
+        values. In this instance the phase change is from sleet to rain."""
+        self.cubes[1].units = Unit('feet')
+        self.cubes[0].rename('altitude_of_rain_falling_level')
+        result = self.plugin.process(self.cubes)
+        expected = np.zeros((3, 3, 3), dtype=np.float32)
+        expected[2] = 1.
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.name(), "probability_of_rain_at_surface")
+        self.assertEqual(result.units, Unit('1'))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_unit_conversion(self):
+        """Test that process returns the same as test_prob_rain when the
+        orography cube units are in feet."""
+        self.cubes[0].rename('altitude_of_rain_falling_level')
+        result = self.plugin.process(self.cubes)
+        expected = np.zeros((3, 3, 3), dtype=np.float32)
+        expected[2] = 1.
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.name(), "probability_of_rain_at_surface")
+        self.assertEqual(result.units, Unit('1'))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_bad_phase_cube(self):
+        """Test that process raises an exception when the input phase cube is
+        incorrectly named."""
+        self.cubes[0].rename('altitude_of_kittens')
+        msg = 'Could not extract a rain or snow falling-level cube from'
+        with self.assertRaisesRegex(ValueError, msg):
+            self.plugin.process(self.cubes)
+
+    def test_bad_orography_cube(self):
+        """Test that process raises an exception when the input orography
+        cube is incorrectly named."""
+        self.cubes[1].rename('altitude_of_kittens')
+        msg = 'Could not extract surface_altitude cube from'
+        with self.assertRaisesRegex(ValueError, msg):
+            self.plugin.process(self.cubes)
+
+    def test_bad_units(self):
+        """Test that process raises an exception when the input cubes cannot
+        be coerced into the same units."""
+        self.cubes[1].units = Unit('seconds')
+        msg = 'Unable to convert from '
+        with self.assertRaisesRegex(ValueError, msg):
+            self.plugin.process(self.cubes)
+
+    def test_spatial_mismatch(self):
+        """Test that process raises an exception when the input cubes have
+        different spatial coordinates."""
+        self.cubes[1] = set_up_variable_cube(
+            self.cubes[1].data, name='surface_altitude', units='m',
+            spatial_grid='latlon', attributes=self.mandatory_attributes)
+        msg = 'Spatial coords mismatch between'
+        with self.assertRaisesRegex(ValueError, msg):
+            self.plugin.process(self.cubes)
+
```

Comment:
Can you add a test for the case I highlighted of units in "m" and "metres"; if it passes, huzzah, if not please make it pass.